js/modules/loki_message_api.js

```diff
@@ -95,19 +97,44 @@ class LokiMessageAPI {
         'Error sending message'
       );
       throw HTTPError('sendMessage: error response', response.status, result);
-    });
-    try {
-      // TODO: Possibly change this to require more than a single response?
-      const result = await firstOf(requests);
-      return result;
-    } catch (err) {
-      throw err;
+    };
+
+    let swarmNodes;
+    while (completedNodes.length < MINIMUM_SUCCESSFUL_REQUESTS) {
```

Comment:
I'm still unsure about keeping track of the completed nodes (either successfully reached or dismissed).
Don't we want to count the _successful_ requests?

Reply:
Well we want to count the dismissed requests as well because otherwise we would never exit this loop (we would just keep sending requests to nodes who are responding with 400 or something, never "succeeding" our request but never removing the nodes from memory because they are still responding)
But you are correct in that currently this has a bug where if we get a non-200 response from a snode when trying to retrieve our messages we will just return thinking we have completed our request even though we haven't really checked for messages
```
completedNodes.push(nodeUrl);

if (response.status === 200) {
  if (result.lastHash) {
    window.LokiSnodeAPI.updateLastHash(nodeUrl, result.lastHash);
    callback(result.messages);
  }
  return;
}
// Handle error from snode
log.error(options.type, options.url, response.status, 'Error');
```